samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates [SnapRules] based on the given asset type, synchronizes the snap sources using these
+     * rules, then updates the snap sources list used by the UI.
+     */
+    private suspend fun setSnapSettings(assetType: UtilityAssetType) {
+        // Get the snap rules associated with the asset type
+        val snapRules = SnapRules.create(arcGISMap.utilityNetworks.first(), assetType).getOrThrow()
+
+        geometryEditor.snapSettings.apply {
+            // Synchronize the snap source collection with the map's operational layers using the snap
+            // rules. Setting SnapSourceEnablingBehavior.SetFromRules will enable snapping for the
+            // layers and sublayers specified in the snap rules.
+            syncSourceSettings(snapRules, SnapSourceEnablingBehavior.SetFromRules)
+
+            // Enable snapping for the graphics overlay as this will not be affected by the given
+            // SnapSourceEnablingBehavior.setFromRules
+            sourceSettings.first { it.source == graphicsOverlay }.isEnabled = true
+        }
+
+        updateSnapSourceList()
+    }
+
+
+    /**
+     * Updates the enabled value of the [SnapSourceSettings] object at the given index and rebuilds
+     * the snap source list.
+     */
+    fun setSnapSourceCheckedValue(checkedValue: Boolean, index: Int) {
+        // Set new value into appropriate property via index
+        _snapSourcePropertyList.value[index].snapSourceSettings.isEnabled = checkedValue
+
+        updateSnapSourceList()
+    }
+
+    /**
+     * Updates the lists used by the UI to show SnapSourceSettings information.
+    */
+    private fun updateSnapSourceList() {
+        // Update the backing list with a new list of current properties from snapSettings
+        _snapSourcePropertyList.value = currentSnapSourcePropertyList()
+    }
+
+    /**
+     * Returns a list of [SnapSourceProperty] objects based on the current snap sources.
+     */
+    private fun currentSnapSourcePropertyList(): List<SnapSourceProperty> {
+        return buildList {
+            geometryEditor.snapSettings.sourceSettings.forEach { sourceSettings ->
+                when (sourceSettings.source) {
+                    is GraphicsOverlay -> {
+                        symbolSwatches[sourceSettings.ruleBehavior]?.let { swatch ->
+                            add(SnapSourceProperty((sourceSettings.source as GraphicsOverlay).id, swatch, sourceSettings))
+                        }
+
+                        // Set the appropriate symbol for the layer based on the SnapRuleBehavior.
+                        graphicsOverlay.renderer = SimpleRenderer(symbols[sourceSettings.ruleBehavior])
+                    }
+                    is SubtypeFeatureLayer -> {
+                        sourceSettings.childSourceSettings.forEach { childSourceSettings ->
+(childSourceSettings.source as? SubtypeSublayer)?.let { childSource ->
```

Comment:
Formatting looks wrong here: 
```suggestion
                            (childSourceSettings.source as? SubtypeSublayer)?.let { childSource ->
```